go.mod

```diff
@@ -15,43 +15,43 @@
 
 module github.com/apache/spark-connect-go/v35
 
-go 1.21
+go 1.22.0
```

Comment:
Is this change needed? Do we need to update the go version at the same time?

Reply:
https://github.com/apache/arrow-go/blob/main/go.mod#L21

we need ~